src/jltypes.c

```diff
@@ -1687,17 +1687,43 @@ jl_value_t *jl_apply_type_(jl_value_t *tc, jl_value_t **params, size_t n)
         }
     }
     if (tc == (jl_value_t*)jl_ntuple_type && (n == 1 || n == 2)) {
-        if (jl_is_long(params[0])) {
-            size_t nt = jl_unbox_long(params[0]);
+        if (!jl_is_typevar(params[0])) {
+            size_t nt;
+            if (jl_is_long(params[0])) {
+                nt = jl_unbox_long(params[0]);
+            }
+#ifdef _P64
+            else if (jl_is_uint64(params[0])) {
```

Comment:
There is no duplication there.  long is either int64 or int32.  the _P64 case catches uint64 and int32, which would otherwise be missed.